packages/common/src/entities/projects/projects.local.ts

```
@@ -267,6 +294,136 @@ export class LocalProjects extends ProjectsAbstract<LocalEnvironment> {
             });
     }
 
+    async listSampleProjects(fetch?: () => any | Got): Promise<List<ISampleProjectRest>> {
+        const get = fetch || got;
+
+        try {
+            const response = await get('https://api.github.com/orgs/neo4j-graph-examples/repos');
+
+            // @TODO: Figure out how to filter the response by topic or something else.
+            return List.from(
+                JSON.parse(response.body).map(({name, description}: ISampleProjectRest) => ({
+                    name,
+                    description,
+                    downloadUrl: `https://github.com/neo4j-graph-examples/${name}/archive/master.zip`,
+                })),
+            );
+        } catch (_error) {
+            throw new FetchError(`Unable to fetch Sample Projects from GitHub`);
+        }
+    }
+
+    async downloadSampleProject(selected: string, destPath?: string): Promise<{path: string; temp: boolean}> {
+        const {tmp} = this.environment.dirPaths;
+
+        const diskPath = destPath || path.join(tmp, 'sample-project');
+        const sampleProject = await download(
+            `https://github.com/neo4j-graph-examples/${selected}/archive/master.zip`,
+            destPath ? destPath : path.join(diskPath, selected),
+        );
+
+        const extractedPath = await extract(sampleProject, diskPath);
+
+        return {
+            path: extractedPath,
+            temp: !destPath,
+        };
+    }
+
+    async installSampleProject(
+        srcPath: string,
+        args: {
+            name?: string;
+            nameOrId?: string;
+            temp?: boolean;
+        },
+    ): Promise<{project: IProjectInput; install?: ISampleProjectInput}> {
+        const {name, nameOrId, temp} = args;
+
+        let project;
+
+        if (nameOrId && !temp) {
+            project = await this.get(nameOrId);
+        } else if (!nameOrId && !temp && name) {
+            project = await this.link(srcPath, name);
+        }
+
+        if (!project) {
```

Should we use the existing project if one already exists, or should we just throw an error? I'm guessing if a user is trying to install a sample project they probably don't want to overwrite an existing one.

---

Well, there are three scenarios here.

1. `projectId` is provided, we just get the project info from the id and return. Desktop does this.
2. A `name` and a `srcPath` exists, this is a folder on disk that should be linked. 
3. None of the above is satisfied, we create a new project from scratch and copy the files there.

---

What is `srcPath`? Is it the path to the extracted sample project or the path where the project should be installed? 

Reading it again, I think I might have misunderstood what this method is doing, it seems like this is not actually installing the sample project, but rather getting/creating/linking the project where the data will be copied and where the DBMSs will be created (?)

I'm still not sure why we would expect to use an existing project when installing a sample project though. Is it just because Desktop is creating it instead?

---

sounds like a great test case?